pkg/actuators/machine/instances_test.go

```diff
@@ -1544,3 +1544,35 @@ func TestGetAvalabilityZoneTypeFromZoneName(t *testing.T) {
 		})
 	}
 }
+
+func TestGetCapacityReservationSpecification(t *testing.T) {
+	mockCapacityReservationID := "cr-123"
+	testCases := []struct {
+		name                  string
+		capacityReservationID string
+		expectedRequest       *ec2.CapacityReservationSpecification
+	}{
+		{
+			name:                  "with no CapacityReservationID options specified",
+			capacityReservationID: "",
+			expectedRequest:       nil,
+		},
+		{
+			name:                  "with a valid CapacityReservationID specified",
```

Comment:
Is there any scenario where we may pass an invalid CapacityReservationID? Do we want some verification or testing around that?

Reply:
All the AWS resources start with their short resource name for example cr-***, so I thought no need for verification.

Reply:
You still need to verify the user input, what if they change the value to something that isn't the correct format?

Reply:
Done, verify that the id starts with the prefix  `cr-`.